plugins/module_utils/net_map/networking_definition.py

```diff
@@ -1008,6 +1008,107 @@ def __init__(
         super().__init__(network, raw_config, self.__OBJECT_NAME)
 
 
+class RouterDefinition:
+    """Parser and validator of a signle router config element
+
+    Handles the parsing and validation of a router item in the
+    dict of routers of the Networking Definition.
+
+    The network definition that this class parses should follow
+    this format:
+
+    .. code-block:: YAML
+
+       <router-name>:
+         external_network: <external gateway network: optional>
+         networks:  <List of networks attached to the router>
+           - <Network 1>
+
+    Raises:
+        ValueError: If name it's not provided.
+        exceptions.NetworkMappingValidationError: When the format
+            of the provided instance is not correct.
+    """
+
+    __OBJECT_NAME = "router"
+
+    __FIELD_EXTERNAL_NETWORK = "external_network"
+    __FIELD_NETWORKS = "networks"
+
+    def __init__(self, name: str, raw_definition: typing.Dict[str, typing.Any]) -> None:
+        """Initializes a RouterDefinition from dict with its parameters
+
+        Args:
+            name: The name of router.
+            raw_definitions: The dictionary that contains the configuration
+                of the router
+
+        Raises:
+            ValueError: If name is not provided.
+        """
+        if not name:
+            raise ValueError("name is a mandatory argument")
+
+        self.__name = name
+        self.__external_network = None
+        self.__networks = []
+        self.__parse_raw(raw_definition)
+
+    @property
+    def name(self) -> str:
+        """Name of the network."""
+        return self.__name
+
+    @property
+    def external_network(self) -> typing.Optional[str]:
+        """External gateway network"""
+        return self.__external_network
+
+    @property
+    def networks(self) -> typing.List[str]:
+        """List of networks to attach to the router"""
+        return self.__networks
+
+    def __parse_raw(self, raw_definition: typing.Dict[str, typing.Any]):
+        if self.__FIELD_EXTERNAL_NETWORK in raw_definition:
```

Comment:
This check is not needed as the `_validate_parse_field_type` function, when `mandatory=False` and the field is not present returns just `None`.